cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -68,6 +69,95 @@ using internal::MakeConverter;
 
 namespace py {
 
+enum class MonthDayNanoField { kMonths, kWeeksAndDays, kDaysOnly, kNanoseconds };
```

Comment:
Since we are at it, can you perhaps enclose all private APIs in the anonymous namespace?

Reply:
i had these in an anonymous namespace originally and got obscure linking errors.  I can try it again to see if I get different results.

Reply:
After moving code around i was able to move the new ones to private.  I'll play around with adding some more.